### PR TITLE
Cloud library Page

### LINKED ISSUE #72: how to use the cloud library is not documented
there's no documentation on how to interact with the cloud library 
 - how to create a login user
 - push
 - pull

### PR DESCRIPTION
**Pull request by WestleyK**
## Description of the Pull Request (PR):

This PR will add a Cloud Library page. This will cover: Making a account, Making a access token, `pull` command, `push` command, and the remote builder

## This fixes or addresses the following GitHub issues:

- Fixes #72

### CODE REVIEW

File: cloud_library.rst [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+This page will cover how to use our cloud services with Singularity.
+
+--------
+Overview
+--------
+
+The Cloud Library is the place to push (upload) your containers to the cloud so other users can
+download, :ref:`Verify <signNverify>` (optional), and use the containers.
+
+We also pervide a :ref:`Remote Builder <remote_builder>`, this is used for building your containers remotly,
```
Comment (jchavezlavalle) [inline cloud_library.rst]: pervide -> provide, remotly -> remotely, advantags -> advantages

File: cloud_library.rst [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+this offers advantags:
+
+ - Usally fast.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: Usually

File: cloud_library.rst [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+ - No root access needed.
+
+ - Can build directly to the library without the need to :ref:`push <push>` your container after. (only thrught the web GUI)
```
Comment (jchavezlavalle) [inline cloud_library.rst]: only **through**

File: cloud_library.rst [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+
+content...
+
+.. _make_a_account:
+
+-------------
+Make a Acount
```
Comment (jchavezlavalle) [inline cloud_library.rst]: an Account

File: cloud_library.rst [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+-------------
+
+Makeing a acount is easy, and straightforward:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: Making an account

File: cloud_library.rst [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+
+ 1. Go to: https://cloud.sylabs.io/library.
+ 2. Click "Sign in to Sylabs" (top right corner).
+ 3. Select you method to sign in, with Google, GitHub, GitLab, or Microsoft.
+ 4. Type your passwords, and thats it!
```
Comment (jchavezlavalle) [inline cloud_library.rst]: that's it

File: cloud_library.rst [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+
+Now that you have your token, you are ready to push your container!
+
+.. _push:
+
+-------------------
+Pushing a Container
+-------------------
+
+The ``singularity push`` will push a container to the container library with the givin URL. Heres an example of a typical push command:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: with the **given** URL. **Here's** an example...

File: cloud_library.rst [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+
+.. code-block:: console
+
+    $ singularity push my-container.sif library://your-name/project-dir/my-container:latest
+
+The ``:latest`` is the contianer tag. Tags are used to have difrent version of the same container. Heres an example:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: container, different, Here's

File: cloud_library.rst [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+
+Lets assume you have your container (v1.0.1), and you want to push that container without deleting your ``:latest`` container, then you can add a version tag to that container, like so:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: Let's

File: cloud_library.rst [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+
+.. code-block:: console
+
+    $ singularity push my-container.sif library://your-name/project-dir/my-container:1.0.1
+
+You can download the container with that tag by replacing the ``:latest``, with the taged container you want to download.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: tagged container

File: cloud_library.rst [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+
+-------------------
+Pulling a container
+-------------------
+
+The ``singularity pull`` will pull a container from the `Library <https://cloud.sylabs.io/library>`_ (``library://``), and also `Docker Hub <https://hub.docker.com/>`_ (``docker://``).
+
+.. note::
+
+    When pulling from Docker, the container will automaticly be converted to a SIF (Singularity Image Format) container.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: automatically

File: cloud_library.rst [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+
+Heres a typical pull command:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: Here's

File: cloud_library.rst [hunk 13]
```diff
@@ -0,0 +1,190 @@
...
+
+.. code-block:: console
+
+    $ singularity pull file-out.sif library://alpine:latest
+
+.. note::
+
+    If theres no tag after the container name, Singularity will pull the container with the ``:latest`` tag.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: If there's

File: cloud_library.rst [hunk 14]
```diff
@@ -0,0 +1,190 @@
...
+
+Pulling your own container
+--------------------------
+
+The ``singularity pull`` can pull a container from the `Library <https://cloud.sylabs.io/library>`_ (``library://``),
+and also `Docker Hub <https://hub.docker.com/>`_ (``docker://``).
+
+.. note::
+
+    When pulling from Docker, the container will automaticly be converted to a SIF (Singularity Image Format) container.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: automatically

File: cloud_library.rst [hunk 15]
```diff
@@ -0,0 +1,190 @@
...
+
+Heres a typical pull command:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: Here's

File: cloud_library.rst [hunk 16]
```diff
@@ -0,0 +1,190 @@
...
+
+.. code-block:: console
+
+    $ singularity pull file-out.sif library://alpine:latest
+
+If theres no tag after the container name, Singularity will pull the container with the latest tag.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: there's

File: cloud_library.rst [hunk 17]
```diff
@@ -0,0 +1,190 @@
...
+
+To pull a container with a spicific tag, just add the tag to the library URL:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: specific

File: cloud_library.rst [hunk 18]
```diff
@@ -0,0 +1,190 @@
...
+
+.. code-block:: console
+
+    $ singularity pull file-out.sif library://alpine:3.8
+
+Of course you can pull your own containers. Heres what that will look like:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: comma after "Of course", also typo in Here's

File: cloud_library.rst [hunk 19]
```diff
@@ -0,0 +1,190 @@
...
+
+.. code-block:: console
+
+    $ singularity pull library://your-name/project-dir/my-container:latest
+
+    # or use a difrent tag:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: typo in different

File: cloud_library.rst [hunk 20]
```diff
@@ -0,0 +1,190 @@
...
+
+    $ singularity pull library://your-name/project-dir/my-container:1.0.1
+
+.. note::
+
+    You dont have to specify a output file, one will be created automaticly, but its good practice to always
```
Comment (jchavezlavalle) [inline cloud_library.rst]: typos : don't, automatically , it's

File: cloud_library.rst [hunk 21]
```diff
@@ -0,0 +1,190 @@
...
+Check out :ref:`this page <signNverify>` on how to: :ref:`verify a container <verify_container_from_library>`,
+:ref:`making PGP key, and sign your own containers <sign_your_own_containers>`.
+
+.. _search_the_library:
+
+------------------------------------
+Searching the Library for Containers
+------------------------------------
+
+When it come to searching the library, you could always go to: https://cloud.sylabs.io/library and search from there.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: When it **comes** to searching the library...

File: cloud_library.rst [hunk 22]
```diff
@@ -0,0 +1,190 @@
...
+Or you can use ``singularity search <container/user>``, this will search the library for the ``<container/user>``.
+
+Using the CLI Search
+--------------------
+
+Here is a example for searching the library for ``centos``:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: an example

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

File: cloud_library.rst [hunk 23]
```diff
@@ -0,0 +1,227 @@
...
+
+---------------
+Make an Account
+---------------
+
+Making an account is easy, and straightforward:
+
+ 1. Go to: https://cloud.sylabs.io/library.
+ 2. Click "Sign in to Sylabs" (top right corner).
+ 3. Select you method to sign in, with Google, GitHub, GitLab, or Microsoft.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: your method

File: cloud_library.rst [hunk 24]
```diff
@@ -0,0 +1,227 @@
...
+Check out :ref:`this page <signNverify>` on how to: :ref:`verify a container <verify_container_from_library>`,
+:ref:`making PGP key, and sign your own containers <sign_your_own_containers>`.
+
+.. _search_the_library:
+
+------------------------------------
+Searching the Library for Containers
+------------------------------------
+
+When it comes to searching the library, you could always go to: https://cloud.sylabs.io/library and search from there.
```
Comment (jchavezlavalle) [inline cloud_library.rst]: add link hyperref here

File: cloud_library.rst [hunk 25]
```diff
@@ -0,0 +1,227 @@
...
+
+Notice there are different tags for the same container.
+
+.. _remote_builder:
+
+--------------
+Remote Builder
+--------------
+
+The remote builder service can build your container remotly, (you dont need root access
```
Comment (jchavezlavalle) [inline cloud_library.rst]: remotely, don't

File: cloud_library.rst [hunk 26]
```diff
@@ -0,0 +1,227 @@
...
+to use remote builder)
+
+Building from a defition file:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: definition

File: cloud_library.rst [hunk 27]
```diff
@@ -0,0 +1,227 @@
...
+------------------------------
+
+This is are defition file, lets call it ``ubuntu.def``:
```
Comment (jchavezlavalle) [inline cloud_library.rst]: definition, let's

**Review by jchavezlavalle — CHANGES_REQUESTED** [review]

File: cloud_library.rst [hunk 28]
```diff
@@ -0,0 +1,229 @@
...
+
+--------
+Overview
+--------
+
+The Cloud Library is the place to :ref:`push <push>` (upload) your containers to the cloud so other users can
+download, :ref:`Verify <signNverify>` (optional), and use the containers.
+
+We also provide a :ref:`Remote Builder <remote_builder>`, this is used for building your containers remotely,
+witch allows you to build containers without root access
```
Comment (jchavezlavalle) [inline cloud_library.rst]: **which** allows you...

**Review by jchavezlavalle — DISMISSED** [review]

**Review by GodloveD — APPROVED** [review]
made a few changes and now it lgtm.  :shipit:

**Comment by WestleyK** [comment]
Thanks @GodloveD for fixing this up!

**Pull request merged**